### PR TITLE
[Merged by Bors] - don't certify if URL or pubkey is empty

### PR DESCRIPTION
**Pull request by poszu**
## Motivation

Fixes the issue when the poet client tries to certify despite the certifier URL is empty.

## Description

The issue is observed on all networks with poet v0.10.6+, because it returns a non-null, but empty certifier config like:
```json
{
  "certifier": {
    "url": "",
    "pubkey": ""
  }
}
```
instead of:
```json
{
  "certifier": null
}
```

This results in the following warning on the node side:

```
{"L":"WARN","T":"2024-11-12T21:00:02.403Z","N":"node.atxBuilder","M":"failed to certify poet","error":"certifying POST at : sending request: POST certify giving up after 1 attempt(s): Post \"certify\": unsupported protocol scheme \"\"","smesherID":"5b7f1f"}
```

This PR adds a check and certifies only if both URL and pubkey are configured (non empty).

## Test Plan

Added a test to cover the scenario when URL or pubkey is empty.

## TODO

- [x] Explain motivation or link existing issue(s)
- [x] Test changes and document test plan
- [ ] Update documentation as needed
- [x] Update [changelog](../CHANGELOG.md) as needed

### CODE REVIEW

**Review by fasmat — APPROVED** [review]

**Comment by jellonek** [comment]
Just to ensure - this is needed only to support poets which do not have yet https://github.com/spacemeshos/poet/pull/543 included?

**Review by jellonek — APPROVED** [review]

**Comment by poszu** [comment]
> Just to ensure - this is needed only to support poets which do not have yet [spacemeshos/poet#543](https://github.com/spacemeshos/poet/pull/543) included?

Kinda, keep in mind that people are free to use custom implementations of poets that might behave like that too.

File: activation/poet.go [hunk 1]
```diff
@@ -647,7 +647,7 @@ func (c *poetService) Certify(ctx context.Context, id types.NodeID) (*certifier.
 		return nil, err
 	}
 
-	if info.Certifier == nil {
+	if info.Certifier == nil || len(info.Certifier.Url.Host) == 0 || len(info.Certifier.Pubkey) == 0 {
```
Comment (ivan4th) [inline activation/poet.go:650]:
```suggestion
	if info.Certifier == nil || info.Certifier.Url.Host == "" || len(info.Certifier.Pubkey) == 0 {
```

Comment (poszu) [reply activation/poet.go:650]: Is there any practical difference?

**Comment by poszu** [comment]
bors merge

**Comment by poszu** [comment]
bors merge

**Comment by fasmat** [comment]
bors merge

**Comment by poszu** [comment]
bors merge

**Comment by poszu** [comment]
bors merge

**Comment by fasmat** [comment]
bors merge

**Pull request closed without merging**